src/test/regress/expected/fkeys_between_local_ref.out

```diff
@@ -0,0 +1,145 @@
+\set VERBOSITY terse
+SET citus.next_shard_id TO 1518000;
+SET citus.shard_replication_factor TO 1;
+CREATE SCHEMA fkeys_between_local_ref;
+SET search_path TO fkeys_between_local_ref;
+SET client_min_messages to ERROR;
+-- create a view for testing
+CREATE VIEW citus_local_tables_in_schema AS
+SELECT logicalrelid FROM pg_dist_partition, pg_tables
+WHERE tablename=logicalrelid::regclass::text AND
+      schemaname='fkeys_between_local_ref' AND
+      partmethod = 'n' AND repmodel = 'c';
+-- remove coordinator if it is added to pg_dist_node
+SELECT COUNT(master_remove_node(nodename, nodeport)) < 2
+FROM pg_dist_node WHERE nodename='localhost' AND nodeport=:master_port;
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+CREATE TABLE local_table_1 (col_1 INT UNIQUE);
+CREATE TABLE local_table_2 (col_1 INT UNIQUE);
+CREATE TABLE local_table_3 (col_1 INT UNIQUE);
+CREATE TABLE local_table_4 (col_1 INT UNIQUE);
+INSERT INTO local_table_1 SELECT i FROM generate_series(195, 205) i;
+INSERT INTO local_table_2 SELECT i FROM generate_series(195, 205) i;
+INSERT INTO local_table_3 SELECT i FROM generate_series(195, 205) i;
+INSERT INTO local_table_4 SELECT i FROM generate_series(195, 205) i;
+--                                        _
+--                                       | |
+--                                       | v
+-- local_table_2 -> local_table_1 -> local_table_4
+--                      ^ |               |
+--                      | v               |
+--                  local_table_3 <--------
+ALTER TABLE local_table_2 ADD CONSTRAINT fkey_1 FOREIGN KEY (col_1) REFERENCES local_table_1(col_1);
+ALTER TABLE local_table_3 ADD CONSTRAINT fkey_2 FOREIGN KEY (col_1) REFERENCES local_table_1(col_1);
+ALTER TABLE local_table_1 ADD CONSTRAINT fkey_3 FOREIGN KEY (col_1) REFERENCES local_table_3(col_1);
+ALTER TABLE local_table_1 ADD CONSTRAINT fkey_4 FOREIGN KEY (col_1) REFERENCES local_table_4(col_1);
+ALTER TABLE local_table_4 ADD CONSTRAINT fkey_5 FOREIGN KEY (col_1) REFERENCES local_table_3(col_1);
+ALTER TABLE local_table_4 ADD CONSTRAINT fkey_6 FOREIGN KEY (col_1) REFERENCES local_table_4(col_1);
+CREATE TABLE reference_table_1(col_1 INT UNIQUE, col_2 INT);
+INSERT INTO reference_table_1 SELECT i FROM generate_series(195, 205) i;
+SELECT create_reference_table('reference_table_1');
+ create_reference_table
+---------------------------------------------------------------------
+
+(1 row)
+
+-- we fall back to legacy behavior when coordinator is not added to metadata
```

Comment:
the comment and the behavior don't match. I think it is expected that `ALTER TABLE` fails.

I think we can consider adding the examples we discussed on the PR for coordinator not added case:
```SQL


create table ref (a int primary key);
select create_reference_table('ref');
create table other (x int primary key, y int);
alter table other add constraint fk foreign key (y) references ref (a) on delete cascade;
select create_reference_table('other');

create table ref (a int primary key);
select create_reference_table('ref');
create table other (x int primary key, y int);
alter table other add constraint fk foreign key (y) references ref (a) on delete cascade;


drop table if exists ref, ref2 cascade;


create table ref (a int primary key);
create table ref2 (x int);
alter table ref2 add constraint fk foreign key (x) references ref (a);
select create_reference_table('ref');
alter table ref2 add constraint fk2 foreign key (x) references ref (a);

ERROR:  cannot create foreign key: table "ref" is already distributed
Time: 2.276 ms
```

Reply:
0bac4344012c9cd19356b6aff2ef6dd0a0540690